### DISCUSSION TITLE
NestJs with Graphql not working

### DISCUSSION THREAD

**Discussion by douglasgomes98** [enisdenjo/graphql-sse#102] [Q&A]
**Screenshot**
Visualising is always helpful.
![Screenshot 2024-07-16 at 11 20 20](https://github.com/user-attachments/assets/07bebfaa-dc5f-493b-ae9c-adc396130c95)
https://codesandbox.io/p/github/douglasgomes98/nest-graphq-sse/main

I created a codesandox to help with reproduction.
By default, NestJs uses express as a server and I'm using Apollo Graphql

**Comment by enisdenjo** [comment]
Hey there. The error is not related to graphql-sse, your subscriptions implementation is not working.

I am not very familiar with NestJS, but can you maybe try changing the subscription publish like this?
```diff
- await pubsub.publish(topic, { id, message });
+ await pubsub.publish(topic, { receiveMessage: { id, message } }); 
```

**Reply by douglasgomes98** [reply]
Done! there really was a problem when publishing the message.
But I still get the same error when executing the subscription.
![Screenshot 2024-07-16 at 13 52 38](https://github.com/user-attachments/assets/6bc7ea76-3015-4ff5-8d24-fae264e1880c)

**Reply by douglasgomes98** [reply]
![Screenshot 2024-07-16 at 14 02 53](https://github.com/user-attachments/assets/76de0bda-61b3-4754-92e7-c99b1dbd9678)

**Reply by enisdenjo** [reply]
Maybe there's something in the [NestJS Subscriptions docs](https://docs.nestjs.com/graphql/subscriptions) that could help.

**Reply by douglasgomes98** [reply]
unfortunately I didn't find anything about it.
